src/mpid/ch4/src/ch4_globals.c

```diff
@@ -107,6 +123,60 @@ int MPID_Abort(MPIR_Comm * comm, int mpi_errno, int exit_code, const char *error
     return 0;
 }
 
+#undef FUNCNAME
+#define FUNCNAME MPIDI_Check_for_failed_procs
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+int MPIDI_Check_for_failed_procs(void)
+{
+    int mpi_errno = MPI_SUCCESS;
+    int pmi_errno;
+    int len;
+    char *kvsname;
+    char *failed_procs_string = NULL;
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_CHECK_FOR_FAILED_PROCS);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_CHECK_FOR_FAILED_PROCS);
+
+    /* FIXME: Currently this only handles failed processes in
+     * comm_world.  We need to fix hydra to include the pgid along
+     * with the rank, then we need to create the failed group from
+     * something bigger than comm_world. */
+#ifndef USE_PMIX_API
+    pmi_errno = PMI_KVS_Get_my_name(kvsname, MPIDI_KVSAPPSTRLEN);
+    if (pmi_errno != PMI_SUCCESS) {
+        mpi_errno = MPI_ERR_OTHER;
+        MPIR_ERR_POP(mpi_errno);
+    }
+#endif
+#ifdef USE_PMI2_API
```

Comment:
No `#endif`?

Reply:
It supposes to be `#elif` up there, but I put `#if` there.

Reply:
Should this be an `#elif` as well? Do we expect to use PMIX and PMI2 at the same time?

Reply:
This one needs to be `#ifdef`. The above section is disabled only when PMIx is used. Both PMI and PMI2 needs that `PMI_KVS_Get_my_name`.